src/arguments.cpp

```diff
@@ -222,6 +223,36 @@ std::tuple<IpVer, std::string, uint8_t> Arguments::asIpAddrMask()
     throw std::invalid_argument(err);
 }
 
+#ifdef SWITCHABLE_ETH1_PORT
+bool Arguments::asEthSrc(const std::optional<std::string>& param)
+{
+    const char* arg;
+    if (!param)
+    {
+        arg = asText();
+    }
+    else
+    {
+        arg = param.value().c_str();
+    }
+
+    if (!strcmp("BMC", arg))
+    {
+        return true;
+
+    }
+    if (!strcmp("Host", arg))
+    {
+        return false;
+    }
+    std::string err = "Invalid argument: ";
+    err += arg;
+    err += ", expected 'BMC' or 'Host' string value. ";
+    err += "Please, enter correct argument.";
```

Comment:
Take a look on `Arguments::asToggle()`. I guess this should be rewritten using `asOneOf()` to avoid code duplicity.